clang/test/Frontend/verify.c

```diff
@@ -157,3 +157,33 @@ unexpected b; // expected-error@33 1-1 {{unknown type}}
 // what-error {{huh?}}
 // CHECK9: error: 'what-error' diagnostics expected but not seen:
 #endif
+
+#ifdef TEST_WIDE_DELIM
+// RUN: not %clang_cc1 -DTEST_WIDE_DELIM -verify %s 2>&1 | FileCheck -check-prefix=CHECK-WIDE-DELIM %s
+
+// expected-error {{{some message with {{}} in it}}}
+// expected-error {{{some message with  {}} in it}}}
+// expected-error {{{some message with {{}  in it}}}
+
+// expected-error-re {{{some {{.*}} regex with double braces}}}
+// expected-error-re {{{some message with {{}  in it}}}
+
+// expected-error {{{mismatched delim}}
+// expected-error-re {{{mismatched re {{.*} }}}
+// expected-error-re {{{no regex}}}
+
+#if 0
+//      CHECK-WIDE-DELIM: error: 'expected-error' diagnostics expected but not seen:
+// CHECK-WIDE-DELIM-NEXT:   verify.c Line 164: some message with {{[{]{}[}]}} in it
```

Comment:
Well, it seems like the answer to "who checks the checker" is FileCheck. There wasn't a better way than this that I saw at that layer of recursion to escape the `{{` braces, hopefully it's not too distracting here.

Reply:
Its more the "#if 0" I'm concerned about, are these checks even running?  I thought FileCheck used the lexer to get preprocessed files.

Reply:
Ah, I hear you: they failed when I didn't get the regex right, so I'm pretty convinced they're running un-pre-processed.

Reply:
Huh, I was convinced I've seen folks disable 'check' lines in the past with preprocessor macros... hmm..